src/terms/operators.jl

```diff
@@ -114,16 +114,23 @@ struct MagneticFieldOperator{T <: Real, AT} <: RealFourierOperator
     kpoint::Kpoint{T}
     Apot::AT  # Apot[α][i,j,k] is the A field in (cartesian) direction α
 end
+
 function apply!(Hψ, op::MagneticFieldOperator, ψ)
     # TODO this could probably be better optimized
+    # The cartesian coordinates should be pre-allocated....
+    p = Gv_to_array(Gplusk_vectors_cart(op.basis, op.kpoint))
```

Comment:
This looks very unusual to me. Why do you need this in the first place?

Reply:
This is because the G vectors are stored as a vector of `SVector{3, T}`. This makes it impossible to use broadcast with slicing that is needed for GPU. On CPU one can do `[elem[1] for elem in Gplusk_vectors_cart(op.basis, op.kpoint)]` to get say the first element of each vector  just fine, but this results in scalar indexing that does not work for GPU. 

The solution here is to reinterpret a vector of `SVector{3,T}` as a dense array, as the underlying memory layout is the same, I think. Then we can use slicing with the dot syntax.

Reply:
I see, but in any case  this is a bit of a hack and I would not endorse this with a function on the level of the `G_vectors(basis)`-family of functions. Also how brittle is this? Reinterpret can (and has lead us) to very subtle and annoying bugs, so I'm a bit hesitant. For sure you should add a comment explaining the precise assumptions about memory layout etc. to the place you are using this code.